### ISSUE TITLE
kube-apiserver responds with 200 with empty body

### ISSUE THREAD

**Issue by localghost** [rancher/rancher#8772]
**Rancher versions:**
rancher/server: 1.5.9
rancher/agent: 1.2.2

**Infrastructure Stack versions:**
healthcheck: 0.2.3
ipsec: v0.11
network-services: rancher/metadata:v0.9.1, rancher/network-manager:v0.6.6
scheduler: rancher/k8s:v1.5.4-rancher1-4
kubernetes (if applicable): rancher/k8s:v1.5.4-rancher1-4

**Docker version: (`docker version`,`docker info` preferred)**
```
Containers: 59
 Running: 43
 Paused: 0
 Stopped: 16
Images: 31
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-253:3-201737766-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 7.748 GB
 Data Space Total: 107.4 GB
 Data Space Available: 30.28 GB
 Metadata Space Used: 12.19 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.135 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: journald
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: overlay bridge null host
Swarm: inactive
Runtimes: docker-runc runc
Default Runtime: docker-runc
Security Options: seccomp
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 4
Total Memory: 7.64 GiB
Name: master-10-9-4-64
ID: KNZG:VTJQ:PUUN:MKOX:P4XS:VXQ3:A5IR:JSFQ:RKSC:6JZN:OHF3:73YN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 docker-registry:5000
 127.0.0.0/8
Registries: docker.io (secure)
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
```
3.10.0-327.el7.x86_64
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
KVM

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes

**Steps to Reproduce:**
1. Remove container with Kubernetes API server (`kubernetes-kubernetes-1` in my case)
2. In parallel, list pods:
```
curl -v http://[RANCHER_SERVER]:8080/r/projects/1a7/kubernetes/api/v1/pods
```
The real problematic case with this error is that it sometimes happens during cluster setup; however, it is easier to reproduce it on already deployed and stable cluster.

**Results:**
Expected: either 200 response with valid json content or 404 (or some other error indication)
Actual: 200 response with empty body

**Comment by loganhz** [comment]
With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.
